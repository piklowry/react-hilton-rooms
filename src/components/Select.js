import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Container = styled.div`
  margin: ${props => props.margin};
  display: flex;
  flex-direction: column;
  width: 50px;
`;

const Label = styled.label`
  line-height: 1.2;
  margin: 0 0 8px 0;
  text-align: center;
`;

const Select = ({
  margin,
  userType,
  ageNum,
  isDisabled,
  selectValue,
  handleSelectChange,
  name,
  roomNum
}) => {
  return (
    <Container margin={margin}>
      <Label
        htmlFor={`select-${userType}-${roomNum}`}
        data-testid={'select-label-tag'}
      >
        {userType}
        <br />({ageNum})
      </Label>

      <select
        name={name}
        id={`select-${userType}-${roomNum}`}
        value={selectValue}
        onChange={handleSelectChange}
        disabled={isDisabled}
      >
        <option value="0">0</option>
        <option value="1">1</option>
        <option value="2">2</option>
        <option value="3">3</option>
      </select>
    </Container>
  );
};

Container.propTypes = {
  margin: PropTypes.string,
};

Select.propTypes = {
  margin: PropTypes.string,
  userType: PropTypes.string.isRequired,
  ageNum: PropTypes.string.isRequired,
  isDisabled: PropTypes.bool.isRequired,
  name: PropTypes.string.isRequired,
  handleSelectChange: PropTypes.func,
  selectValue: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
};

export default Select;
