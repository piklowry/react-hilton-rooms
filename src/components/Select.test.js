import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Select from './Select.js';

afterEach(cleanup);

it('renders correctly', () => {
  const { asFragment } = render(
    <Select
      userType={'Adult'}
      isDisabled={false}
      ageNum={'18'}
      name={'adult'}
    />
  );
  expect(asFragment()).toMatchSnapshot();
});



it('inserts text in label tag', () => {
  const { getByTestId } = render(
    <Select
      userType={'Adult'}
      isDisabled={false}
      ageNum={'18'}
      name={'adult'}
    />
  );

  expect(getByTestId('select-label-tag')).toHaveTextContent('Adult');
});
