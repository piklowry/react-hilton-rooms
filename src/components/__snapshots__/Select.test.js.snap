// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa bTEbsp"
  >
    <label
      class="sc-bwzfXH veGOu"
      data-testid="select-label-tag"
      for="select-Adult-undefined"
    >
      Adult
      <br />
      (18)
    </label>
    <select
      id="select-Adult-undefined"
      name="adult"
    >
      <option
        value="0"
      >
        0
      </option>
      <option
        value="1"
      >
        1
      </option>
      <option
        value="2"
      >
        2
      </option>
      <option
        value="3"
      >
        3
      </option>
    </select>
  </div>
</DocumentFragment>
`;
